File: CLASE II - TDA/clase_2.1_ayed/clase_1_ayed/c/carton.c


#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

#include "funciones.h"
#include "carton.h"





CartonPtr crearCartonTeclado(); //Comprador seleccione los numeros

CartonPtr crearCarton2(int v[8], char d[15]);

CartonPtr crearCartonAutomatico(char d[15]){

    CartonPtr c = (CartonPtr) malloc(sizeof(struct CartonE));

    strcpy(c->direccion, d);



    //Valores -1 para mejorar la busqueda
    cargarVectorIntConValor(c->vec, 8, -1);

    //puse 10 en lugar de 99 para verificar repetdos mas facil
    cargarVectorIntAleatorioSinRepetir(c->vec, 8, 0,10);



    return c;

};

void destruirCarton(CartonPtr c){
    free(c);
};

void mostrarCarton(CartonPtr c){

    printf("\n\t---------- CARTON ---------\n");
    printf("\tDIRECCION: %s\n\t---VECTOR---\n\t", c->direccion);
    //mostrarVector
    for (int i =0 ; i<8; i++){

        printf(" %d ", c->vec[i]);

    }



};

File: CLASE II - TDA/clase_2.1_ayed/clase_1_ayed/golang/carton.go
package main

import "fmt"

type Carton struct {
	vec       [8]int
	direccion string
}

func crearCartonAutomatico(direccion string) *Carton {
	c := &Carton{direccion: direccion}

	tmp := c.vec[:]
	cargarVectorIntConValor(tmp, -1)
	// En el C: se usa 0..10 para facilitar ver repetidos.
	cargarVectorIntAleatorioSinRepetir(tmp, 0, 10)

	return c
}

func mostrarCarton(c *Carton) {
	fmt.Printf("\n\t---------- CARTON ---------\n")
	fmt.Printf("\tDIRECCION: %s\n\t---VECTOR---\n\t", c.direccion)
	for i := 0; i < 8; i++ {
		fmt.Printf(" %d ", c.vec[i])
	}
}

